lib/rspec/core/notifications.rb

```diff
@@ -496,7 +496,8 @@ def duplicate_rerun_locations
     # @attr duration [Float] the time taken (in seconds) to run the suite
     # @attr examples [Array<RSpec::Core::Example>] the examples run
     # @attr number_of_examples [Fixnum] the number of examples to profile
-    ProfileNotification = Struct.new(:duration, :examples, :number_of_examples)
+    # @attr profiler [RSpec::Core::Profiler] the profiler used to capture examples
+    ProfileNotification = Struct.new(:duration, :examples, :number_of_examples, :profiler)
```

Comment:
I don't think the profiler formatter needs to be exposed of  profile notification.  Seems like once you merge the example/example group event handling into the existing profiler formatter this will no longer be needed on the notification object anymore, right?